api/app/db/crud/weather_models.py

```diff
@@ -476,3 +476,12 @@ def refresh_morecast2_materialized_view(session: Session):
     logger.info("Refreshing morecast_2_materialized_view")
     session.execute(text("REFRESH MATERIALIZED VIEW morecast_2_materialized_view"))
     logger.info(f"Finished mat view refresh with elapsed time: {datetime.datetime.now() - start}")
+
+
+def get_previous_prediction_model_run(session: Session, prediction_model_run: PredictionModelRunTimestamp):
+    """ Get the prediction model run that ran immediately prior to the prediction model run passed as a parameter. """
+    return session.query(PredictionModelRunTimestamp).\
+        filter(PredictionModelRunTimestamp.prediction_model_id == prediction_model_run.prediction_model_id).\
+        filter(PredictionModelRunTimestamp.id < prediction_model_run.id).\
```

Comment:
Do we rely on primary key id ordering here? Would it be better to compare run timestamps?

Reply:
Technically I suppose we do, but prediction model run timestamp records do get added sequentially. I'll change to a != operator for clarity.

Reply:
Does this work:

```
PredictionModelRunTimestamp.run_timestamp < prediction_model_run.run_timestamp
```

If so, that'd be super clear and would be robust against primary key changes.

Reply:
Done!